TogetherCrew: Community Health Check

katerinabc opened the pull request:
#### Project Abstract

If Web3 is all about communities, how do we know if we’re heading in the right direction? We can guess or use anecdotal evidence. We have taken the science-route and are building a community health dashboard that provides science-based recommendation for members and moderators to understand and get a bird's eye view of their community. 

We are proposing a project to:

- develop a framework for Community Health with actionable metrics
- create an open source data collection tool
- implement the tool in Polkadot and Kusama communities, and the wider ecosystem 
- analyse the interactions and perceptions data to validate the framework and provide insights to advance the communities

We have already conducted five community health checks, and are seeking grant-matching from different communities. 

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] A BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (aUSD/USDT) address for the payment of the milestones is provided in the application.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/w3f/Grants-Program?pullRequest=1482) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/w3f/Grants-Program?pullRequest=1482) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/w3f/Grants-Program?pullRequest=1482) it.</sub>

Noc2 reviewed (changes requested):
Thanks for the application. It sounds interesting. I will mark it as ready for review so that others can also look at it. However, I have a few initial questions: 

- Where can I find the results from other grants that you have done so far? 
- Is the front-end open source that you have developed in the past? 
- I’m not sure it makes sense to only focus on Discord to check the health of a community. For example, Polkadot and Kusama always used Elements for most of the discussions in the past. Additionally, ideally, you would integrate on-chain metrics (treasury proposals, voting activity, etc.) and governance tools like  https://polkassembly.io/

katerinabc commented:
Thanks @Noc2 for your questions.

>     * Where can I find the results from other grants that you have done so far?
MetaGame, while not providing a grant, [talked](https://twitter.com/MetaFam/status/1603818967528718336) about the community health check. 

On the [Aragon forum](https://forum.aragon.org/t/real-time-community-health-analytics/3472/25) we shared the research paper, but after we did the community health check, things changed and it was not published. 

At Near, we were able to work with 2 communities (Veridia, and Cannabis Genom DAO). [Cannabis Genom DAO](https://cannabisgeno.me/) published the health check in their Discord. Regarding Near, business development took some time and we are only now beginning with the health check.


>     * Is the front-end open source that you have developed in the past?
Yes, the front-end is open source. The code is in our repository. 

>     * I’m not sure it makes sense to only focus on Discord to check the health of a community. For example, Polkadot and Kusama always used Elements for most of the discussions in the past. Additionally, ideally, you would integrate on-chain metrics (treasury proposals, voting activity, etc.) and governance tools like  https://polkassembly.io/
You mention two points here: First, the community exists in different spaces, and secondly, the health of the community depends on more than interactions/conversations. 

Regarding the first point, I agree with you and have mentioned this in the proposal. I saw that there is an integration between Element and Discord with posts being cross-posted. Assuming that this cross-posting works well, focusing on Discord will still be viable. We will have to go back to our data analysis script to ensure correct identification of members, assuming that some members are active on both tools. 

Regarding the second point, using on-chain data is one of the strategic direction we could go. We are currently discussing two directions: Integrating on-chain data or gpt3. For example, some communities have language channels. For moderators who do not understand the language they are black box. gpt3 could provide translated summaries of what is going on. Another way gpt3 could be beneficial is to show specifically how members build on each others expertise and innovation to come up with new ideas or make decisions. We have also explored and done some test on how gpt3 could help with onboarding.

What we'll do will depend on our expertise (governance, network analysis) and opportunities to partner with others who complement our strength. 

I'd love to here your opinion regarding on-chain data vs gpt3.

randombishop reviewed (changes requested):
Hi @katerinabc and thank you for submitting this application!
First remark: the PR has 2 files with 2 different variations for your proposal, can you please fix the PR to have only one official proposal?
I also have some questions:
- I understand you will need to collaborate with the Discord admins at Polkadot and Kusama servers? Did you already reach out to them already?
- Installing a discord bot with that level of access in a discord server can be tricky, how do you guarantee the safety of your bot? Is its code already open source at the moment?
- Where does the bot publish the data? Does it go to a server of yours or does the discord admin need to manage their own database server? Also a security question here: how do we guarantee the safety of data about all discord users and their chat interactions, which can sometimes be considered sensitive personal information? Especially if pulled from channels that are not 100% public (token gated, particular community members, etc.)
- Finally, in agreement with @Noc2 's point, I like your project idea but in my opinion it would be safer and with less blockers if you applied your approach to data that comes with public APIs and 100% free to use without permission: example on-chain interactions, governance, twitter feeds, etc. Would you consider redesigning your project in a permissionless approach rather than pulling data from Discord?
Thank you!

katerinabc commented:
@randombishop thanks for your comments. Regarding the two files, one is the proposal text and the other a figure that is placed in the proposal. I can remove the figure and share it via a link if this is the preferred approach. 

> * I understand you will need to collaborate with the Discord admins at Polkadot and Kusama servers? Did you already reach out to them already?
I have not reached out to the admins. I have talked with a moderator to understand how useful the health check would be for them, as moderators are our primary target group. 
> 
> * Installing a discord bot with that level of access in a discord server can be tricky, how do you guarantee the safety of your bot? Is its code already open source at the moment?
The code is already open-source. This is the relevant repo: Data fetching: https://github.com/RnDAO/tc-botComm

At the current stage of our project, we do custom health checks for communities, while building the interactive dashboard. While the advantage is that the analysis is not real time, the advantage is a close communication with our data analysts (Ene and me). At this stage, the extracted data is not stored on any server, but analyzed locally. Once we have the anonymized metrics and delivered the report, the raw data is deleted. We're working with LexDAO to make sure we are GDPR compliant. 
 

> * Where does the bot publish the data? Does it go to a server of yours or does the discord admin need to manage their own database server? Also a security question here: how do we guarantee the safety of data about all discord users and their chat interactions, which can sometimes be considered sensitive personal information? Especially if pulled from channels that are not 100% public (token gated, particular community members, etc.)

Currently, the data is not published anywhere, but the person who does the data export (normally I do this for clients) receives a csv file, which is then fed into the data analysis script. What channels are analyzed is up to the community. As mentioned above, the fetched data is analyzed, added to a static dashboard and then shared. Once the work is done, the data is deleted. 

In the future (planned next month), the data will be stored on our server. Beyond that we are considered a decentralized data storage solution or have admins manage their own database. 

To be clear, this proposal is for a customized community health check. Thus, data will be stored locally and deleted once the dashboard is done. You'll get access to the data analyst who have done or are just completing a PhD in network science. We do fun things like running simulations (e.g., what happens if these two CMs start or stop talking with each other? How does the health score change if the community suddenly increases or decreased by 10%?) or digging deeper into specific channels. Both are features that will not exists in our dashboard app for some time. 

> 
> * Finally, in agreement with @Noc2 's point, I like your project idea but in my opinion it would be safer and with less blockers if you applied your approach to data that comes with public APIs and 100% free to use without permission: example on-chain interactions, governance, twitter feeds, etc. Would you consider redesigning your project in a permissionless approach rather than pulling data from Discord?

We are working on integrating on-chain data. We have started to sketch out what data could give us insights into community health and how it could augment the insights we are getting using Discord data. Discord data is incredible rich as it provides insights into the social layer of a community and we do not want to move fully away from it. Thus, permissionless data is on our roadmap. 

Would you consider doing a call to discuss the idea of on-chain and public data for measuring community health in greater detail?


Yes, we would be considering this. This is something we have been discussing internally already. Do you mind having a call/async conversation to help us better shape the direction.

randombishop commented:
@katerinabc thank you for your feedback.
Don't worry about the 2 files, it was my mistake, i thought there were 2 `md` files.
And yes, happy to jump in a call, please send me an email at nabil@web3.foundation with some time slots that work for you... I am in EST timezone (GMT-5)
Thank you!

katerinabc commented:
> the PR has 2 files with 2 different variations for your proposal, can you please fix the PR to have only one official proposal?
@randombishop Does the change requests label apply only to the confusion about the two files? 

Also, I'm noticing that my reply to your first question (reached out to discord admin's) was formatted as a quote. Not sure  if you missed it. I've talked with one community moderator, but happy to have conversations with others (discord admins and community moderators) in the Polkadot/Kusama ecosystem. I'm especially interested about blockers from admins and pain points from community moderators.

randombishop commented:
@katerinabc thanks for the precisions.
- I removed the `changes requested` tag as it was indeed by mistake.
- I am personally convinced by the value of your proposal, but with the following reserves: 
  * i) Validation by the admins of the discord servers (i will try to find out if we have someone responsible for that at W3F and 
  invite them to the planned meeting; please also invite the moderators you already talked with if possible.)
  * ii) Some guarantees that there are no legal or safety issues with exporting chat data (especially for channels that are not 100% public) to CSV files that will be treated by your data scientists.

katerinabc commented:
I've send the community moderator a DM with the link to the grant application and an invitation to join. 

Regarding point ii I'll say its best to talk about it to understand what safety measures you are looking for

katerinabc commented:
@Noc2 and @randombishop 

1. While I see there is potential working with public data (governance data or Twitter), I see this as a greater risk for your side as we will have to develop a new data pipeline and metrics. Working with Discord data benefits you in that the data pipeline has already been build and part of the dashboard has already been tested with users using fake data. This means that we can deliver on the milestones quicker. You will receive custom health checks for the Polkadot and Kusama discord communities. This involves direct access to our data scientists.

2. We are working with lawyers to make sure we are following GDPR rules. The documents (privacy policy, terms of services) are nearly ready. We have received a list of required and recommended actions. Finally, we have a GDPR sessions planned for our team next week.

randombishop commented:
Thanks for the update @katerinabc !
Forwarding the GDPR section to legal for review...

keeganquigley commented:
Hi @katerinabc thanks for the application. Please see the following questions/comments:

- Our [payment options](https://github.com/w3f/Grants-Program#pencil-process) don't include ETH, can you please see the document and then change it [here](https://github.com/w3f/Grants-Program/pull/1482/files#diff-e0e38fd7ff8443c41d9f1e733b33c6135c30c150c1fda0e3bf32a25e424d3963R4)?
- Why not go after the permissionless data first? One unique thing about our ecosystem is that all our governance is on-chain. I'd love to see more tools to analyze the on-chain data.
- I'm not sure that Discord on its own is an accurate representation of the community, as many core contributors don't use it. 
- In my experience, there is more "noise" on Discord than on other platforms such as Element (Matrix protocol), Polkadot forums, Twitter, Reddit, etc. how would this factor into the data?
- A Discord moderator I talked to shared concerns over analyzing our Discord servers, as that is a lot of data that would be able to be collected. Mods would be concerned about what the data contains and is used for. Can you explain more about  what data will be collected and what useful metrics would come out of it?

katerinabc commented:
Hi @keeganquigley, thanks for taking the time to review the application. 
 
> Our [payment options](https://github.com/w3f/Grants-Program#pencil-process) don't include ETH, can you please see the document and then add the currency [here](https://github.com/w3f/Grants-Program/pull/1482/files#diff-e0e38fd7ff8443c41d9f1e733b33c6135c30c150c1fda0e3bf32a25e424d3963R4) after the address?

I've added the currency (USDC).

>   Why not go after the permissionless data first? One unique thing about our ecosystem is that all our governance is on-chain. I'd love to see more tools to analyze the on-chain governance data, for example.

We consider communication between contributors a key aspect of community. When two people communicate, on the surface, they exchange information by sharing opinions/links/ideas/feedback. On a deeper level, they are also developing a mental model, a picture, of each other (can I trust the person? does the person know his/her stuff? Is the person funny? Reliable?). Over time, [these single interactions between two people form a pattern](https://www.researchgate.net/publication/259642505_Short-term_and_long-term_stability_in_electronic_communication_networks), like a signature or a thumb print. Based on this signature or thumb print, the health of the community can be inferred. For this reason, we began with Discord data instead of on-chain governance data. 

We know that other communities "live" on Twitter. Similar to Discord, this is communication data. However, with Twitter there is no clear-cut boundary of who is a community member. It could be followers, but that is a very loose definition (following someone requires less emotional investment than joining a server). For the analytical method we use (social network analysis) it is important that we can clearly say who is part of the community or not. In addition, Twitter's API has other disadvantages that Discord doesn't have (only go back 7 days, rate limit). 

We have started to discuss how to integrate governance data at the end of last year (December 2022). Thus, on-chain data will be included in the health assessment in the future. 


>  I'm not sure that Discord on its own is an accurate representation of the community, as many core contributors don't use it.

I've seen that there is a bridge between Element and Discord, hence I grouped these two platforms into one. I seen them as two room within one house. If Polkadot/Kusama has more "houses" (e.g., Twitter, Polkadot forums, Twitter, Reddit) the accuracy of the health assessment for the overall community will be limited. Depending on the demographics and how each platform is used, the result from Discord could be generalized to other platforms. For example, if only European women use Discord and male US citizens use Reddit, measuring the the health of the community on Discord and assuming it will be the same on Reddit doesn't make sense. But if the member profiles are similar, then focusing on one platform and extrapolating the results to another platform is ok. 

> In my experience, there is more "noise" on Discord than on other platforms such as Element (Matrix protocol), Polkadot forums, Twitter, Reddit, etc. how would this factor into the data?

I'm going to assume that with "noise" you mean spam message, gm/gn postings, and scammers. Our data point isn't a message but an interaction. By doing this a lot of messages are excluded from the analysis. Please let me know if I made a wrong assumption and clarify what you mean with noise.

> A Discord moderator I talked to shared concerns over analyzing our Discord servers, as that is a lot of data that would be able to be collected. Mods would be concerned about what the data contains and is used for. Can you explain more about  what data will be collected and what useful metrics would come out of it?

According to GDPR guidelines only data that is used for analysis and provides value to customers should be collected. We are following this guideline. From a message that is posted in Discord we collect information about the sender, timestamp, the content, emoji reactions, people who were mentioned, if the message is a reply to another message. This is being anonymized, and a network is created. This network is a nXn matrix (n = members) where n(ij) is the number of interactions from member _i_ to member _j_. We then calculate the following metrics:
- Spread of Influence (technical name: centralization)
- Fragmentation (technical name: clustering)
- Cliques-Bridge ratio (technical name: small world)

Beyond that we have developed metrics for engaged and disengaged members based on their level of interaction with others. We distinguish between a member who posts every day but receives no replies/emoji reaction and a member who posts every day and receives replies/emoji reactions. Feel free to have a look at [this prototype](https://www.figma.com/proto/jflSBwHW3b5KgohsaYExuN/TogetherCrew?node-id=1%3A49&starting-point-node-id=1%3A49). The health assessment always contains a handout walking the community through each metric. We also always offer a debriefing (video call).

randombishop reviewed (changes requested):
Hi @katerinabc,
Sorry to let you know that legal department opinion is negative about using private channels data.
But like I said during our meeting, I am willing to go ahead with the project if you refactor it to use public channels and in-chain data; especially if you include polkassembly, matrix and twitter as well.

katerinabc commented:
> 

thanks. Could you specify what is meant with private channels data? Role or token gated discord channels? Or all Discord data.

I'm gonna go back to the team to discuss.

randombishop commented:
@katerinabc not a lawyer so I can't really draw the line; but I'd say anything that is public, not token nor role gated, and can be pulled without any illegal scraping, only using legit APIs, in-chain data, and officially public datasets would be fair game.

randombishop commented:
Hi @katerinabc thanks for refactoring the application, in this new form the project doesn't have any blockers imo.
I would request a couple of changes though: for milestone 2 you state that the code doesn't require tests nor docker deliverable; but we generally require these as best practices. Why not include some tests and a way to run as a docker image in your codebase? (of course only to the applicable parts like the backend data pipeline or something?)

katerinabc commented:
> Hi @katerinabc thanks for refactoring the application, in this new form the project doesn't have any blockers imo. I would request a couple of changes though: for milestone 2 you state that the code doesn't require tests nor docker deliverable; but we generally require these as best practices. Why not include some tests and a way to run as a docker image in your codebase? (of course only to the applicable parts like the backend data pipeline or something?)

Thanks for your comment. I've modified milestone 2 to include tests + test guide for the data fetching and analysis, and a docker file.

randombishop commented:
Thank you for the changes @katerinabc !
I am happy to go ahead with the application now and tagging it as ready to review for the rest of the committee.

Noc2 reviewed (changes requested):
Thanks a lot for the application. I'm generally happy to approve it, but I have one request: Could you integrate the programming language/framework of the dashboard into the specification of the milestone? For example, will you use React or something similar for it?

takahser commented on applications/community-health-check.md:
```diff
@@ -0,0 +1,281 @@
+
+Although our initial focus is community health metrics, the proposed approach sets the foundation for further applications. The combination of SNA and pulse surveys offers unprecedented actionable insights in real-time. Some of the potential applications and insights for the ecosystem are
+
+- Generate baseline metrics for community health / vibez to quantify and better understand the impact a specific event is having on a community and/or sub-groups within the community
```
> vibez

What do you mean by that?

katerinabc replied:
with vibez we mean the feeling/quality of the community

takahser replied:
ah you mean _vibes_

takahser commented on applications/community-health-check.md:
> but technically also rests on individual posting behavior and not interaction data

Could you further elaborate on the difference between _individual posting behavior_ and _interaction data_? Given a network graph where the nodes represent the SN's participants and the edges between them their relationships (e.g. likes, comments, follows, etc.), does it mean that the further only focuses on the nodes while your solution would additionally take the edges into account?

katerinabc replied:
Yes you said it well. Individual posting behavior focuses only on the number of activities of a person, but not how these activities impact others. Edges are our interaction data (replies, mentions, emoji reactions)

takahser replied:
Thanks. If that's the case, could you please incorporate this information into the doc?

takahser commented on applications/community-health-check.md:
```suggestion
- We will prepare a record of processing activities
```

takahser commented on applications/community-health-check.md:
> Given other requests we have received, we have decided that building an integration and health metrics for Twitter is a more viable path for us.

I'm curious to learn what led to the decision to pick Twitter over Reddit.

takahser replied:
@katerinabc ping

takahser commented on applications/community-health-check.md:
Could you specify the license you're planning to use?

takahser replied:
@katerinabc this is still not specified. Please choose one of the [supported licenses](https://github.com/w3f/Grants-Program/#guidelines):

> We prefer Apache 2.0, but GPLv3, MIT or Unlicense are also acceptable.

If you're undecided, you can specify multiple ones and add a note, that you will pick one of them

katerinabc replied:
Apologies. I worked for a long time in academia and there it's just a given that every written document can be re-used by others just with referencing. I'm gonna add multiple ones and decide later.

takahser commented on applications/community-health-check.md:
You can remove these if you're not planning to deliver anything.

katerinabc replied:
Thanks. I'll  leave them in so that it is clear that this is not an error of omission.

takahser replied:
@katerinabc see my latest suggestion on that block.

takahser commented on applications/community-health-check.md:
Could you add more details here?

katerinabc replied:
I've added more information. If you are looking for something else, let me know.

takahser replied:
Thanks. Could you please add more info on which relationships between the nodes (i.e. edges) you're planning to analyse? Ideally, provide a full list of supported edges so we can easily verify the milestone, once delivered.

katerinabc replied:
Please have a look at milestone 1. I have specified how we are building the network by adding the following information:

We will build a directed network, where nodes are always Twitter users. From a network assembly perspective, we will not differentiate between accounts representing people and those representing communities or organizations. The edges between an user profiles are either a reply, quote, mention, retweets, or likes. Thus, a tie from user A to user B exists if (1) user A replies to user B, user A quotes user B, user A mentions user B, user A retweets user B, or user A likes user B's tweet. At this moment, we will created weighted edges, not making a conceptual difference between the interaction type (reply, quote, mention, retweets and likes).

takahser commented on applications/community-health-check.md:
```suggestion
| 0a. | License | The code for data fetching and storage is already available in a public GitHub folder (Unlicensed). We will specify an OSS license. |
```

Could you add a link to the GitHub repo since it's already available?

takahser commented on applications/community-health-check.md:
```suggestion
| 0c. | Testing and Testing Guide | Running the data pipeline and analyzing the data will be covered by tests to ensure functionality. We will describe how to run the tests|.
```

takahser commented on applications/community-health-check.md:
```suggestion
| 0d. | Docker | We will deliver a docker file to tests the functionality. |
```

takahser commented on applications/community-health-check.md:
- Will you write the integration for a paid version of the API or the free tier?
- Can you elaborate on the throughput that this your crawler will be able to deliver (e.g. how many requests/min, how many tweets, comments, likes, etc. can be retrieved by some unit of time).
- How about other accounts, e.g. parachains, dApp developers and other teams that build on Substrate?

katerinabc replied:
The updated proposal will have more information.

takahser replied:
Similarly to the research deliverable in M1, can you please add more specific information about which edges will be supported?

katerinabc replied:
I have decided to not repeat the same information here than for Milestone 1. Milestone 1 is our research phase, while milestone 2 is the implementation. Hence, decisions we are making during milestone 1 will impact milestone 2.

takahser replied:
@katerinabc lovely, thanks for adding that info.

takahser commented on applications/community-health-check.md:
```diff
@@ -0,0 +1,281 @@
+
+Social network data and its methodology (Social Network Analysis; SNA) is a structured way to visualize how communications, information, and value creation occur through an organization based on interaction graphs. Although a relatively new decision-making tool for leaders, SNA is gaining popularity over traditional survey tools. SNA has been linked to a wide range of insights, such as improving  ...
+
+![Data points for measuring community health](howmeasurecommhealth.jpg)
```
```suggestion
![Data points for measuring community health](https://user-images.githubusercontent.com/5393704/224280603-03e8b7a7-4212-4a46-b957-ef06ebe3de5a.png)
```

Ideally, we avoid polluting the `applications` directory with images and reference them from an external URL instead.

takahser reviewed (changes requested):
@katerinabc thanks for submitting, this looks interesting. However, I have a couple of questions, feel free to have a look.

Noc2 reviewed (changes requested):
Just a reminder regarding my question above, and feel free to ping me here once it's updated.

takahser commented on applications/community-health-check.md:
```diff
@@ -0,0 +1,281 @@
+
+Although our initial focus is community health metrics, the proposed approach sets the foundation for further applications. The combination of SNA and pulse surveys offers unprecedented actionable insights in real-time. Some of the potential applications and insights for the ecosystem are
+
+- Generate baseline metrics for community health / vibez to quantify and better understand the impact a specific event is having on a community and/or sub-groups within the community
```
```suggestion
- Generate baseline metrics for community health / vibes to quantify and better understand the impact a specific event is having on a community and/or sub-groups within the community
```

takahser reviewed (changes requested):
Thanks for the replies. However, there are still plenty of open questions of my [previous review](https://github.com/w3f/Grants-Program/pull/1482#pullrequestreview-1334461429).

katerinabc commented:
> Thanks a lot for the application. I'm generally happy to approve it, but I have one request: Could you integrate the programming language/framework of the dashboard into the specification of the milestone? For example, will you use React or something similar for it?

I have updated the file to include the programming language for milestone 2.

katerinabc commented:
@noc2: Thank you for the ping. I missed your question and made the required specification. 

@takahser Thank you for your patience and throughout review. I've done the following
- all minor changes (e.g., typos) you suggested have been accepted.
- removed the figure.
- Milestone 1: I've specified the output we create. If information is missing, let me know. 
- Milestone 2: I've added the license and information about the API. I've also added the github repos where relevant.

takahser commented on applications/community-health-check.md:
```suggestion

Milestone 1 will be focused on research. Hence, we're not going to deliver the following usually mandatory deliverables:

- Documentation
- Testing and Testing Guide
- Docker

| Number | Deliverable | Specification |
| -----: | ----------- | ------------- |
| 0a. | License | The results will be published open-access (no paywall or copyright). |
| 0b. | Article | We will publish an article (technical document) describing the metrics, insights (so-what), and limitations. This article will also explain how the Twitter community is built (e.g., who are nodes, when there is an edge between two people, who is excluded/included, and why). The article will not be behind a paywall. The article will be written for an audience comfortable with data analysis. |
```

takahser replied:
I still think it's easier to skip the deliverables that you're not delivering. This way, it will also be easier to validate the results.

takahser reviewed (changes requested):
@katerinabc thanks, looks much better now. Nevertheless, I added some comments. Mainly, I think it'd be nice to know which edges you're going to research and support.

takahser reviewed (changes requested):
@katerinabc thanks for the update. Just as a reminder, the information on which edges you're going to research and support is still missing, see [this](https://github.com/w3f/Grants-Program/pull/1482#discussion_r1141857607) and [this](https://github.com/w3f/Grants-Program/pull/1482#discussion_r1141871727) comments in the previous threads.

katerinabc commented:
> @katerinabc thanks for the update. Just as a reminder, the information on which edges you're going to research and support is still missing, see [this](https://github.com/w3f/Grants-Program/pull/1482#discussion_r1141857607) and [this](https://github.com/w3f/Grants-Program/pull/1482#discussion_r1141871727) comments in the previous threads.

Hi @takahser I've replied to the comments. I've decided to not copy-paste the information about how edges are created from milestone 1 to milestone 2. I see Milestone 2 as dependent on the outcome of milestone 1.

takahser reviewed (approved):
@katerinabc thanks for the update, lgtm now. 👍

keeganquigley commented on applications/community-health-check.md:
Here is where I'm suggesting to update with the full Apache 2.0 text instead of just the header. Thanks!

katerinabc commented:
@keeganquigley It was always my assumption that milestone 2 will be technical delivery and hence include data about the Polkadot and Kusama twitter community. Given the confusion, I should have communicated this better. 

I was wondering if we could have a call to discuss details of milestone 2, to make sure we are aligned on what will be delivered.

keeganquigley commented:
Hi @katerinabc sure thing, happy to discuss milestone 2 with you. Feel free to book a meeting; I just emailed you with my calendar link. Thanks!

keeganquigley commented:
Hi @katerinabc how is milestone 2 coming along?

katerinabc commented:
Hi @keeganquigley We're user testing the Twitter dashboard right now.

keeganquigley commented:
Hi @katerinabc how is milestone 3 coming along?

katerinabc commented:
We will most probably not deliver milestone 3 within the proposed timeframe given our current roadmap.

keeganquigley commented:
Thanks @katerinabc does that mean you just need a few more weeks or should we go ahead and cancel it and you can pick it up later if you decide you want to work on it.

keeganquigley commented:
I will go ahead and close it but let us know if you want to pick it back up at a later time and we can potentially re-open it.
